pkg/segment/structs/evaluationstructs_test.go

```diff
@@ -1779,3 +1779,121 @@ func Test_StatisticExpr(t *testing.T) {
 	assert.Equal(t, bucketResult2, results[0])
 	assert.Equal(t, bucketResult1, results[1])
 }
+
+func TestFormatTime(t *testing.T) {
+	cases := []struct {
+		time     time.Time
+		format   string
+		expected string
+	}{
+		{
+			time:     time.Date(2023, 3, 14, 1, 59, 26, 0, time.UTC),
+			format:   "%Y-%m-%d %H:%M:%S",
+			expected: "2023-03-14 01:59:26",
+		},
+		{
+			time:     time.Date(2020, 12, 31, 23, 59, 59, 999999000, time.UTC),
+			format:   "%Y-%m-%d %I:%M:%S %p %f",
+			expected: "2020-12-31 11:59:59 PM .999999",
+		},
+		{
+			time:     time.Date(1999, 1, 1, 15, 0, 0, 0, time.UTC),
+			format:   "%A, %B %d, %Y",
+			expected: "Friday, January 01, 1999",
+		},
+		{
+			time:     time.Date(2023, 3, 14, 1, 59, 26, 0, time.UTC),
+			format:   "%F %T",
+			expected: "2023-03-14 01:59:26",
+		},
+		{
+			time:     time.Date(1970, 1, 1, 0, 0, 0, 0, time.UTC),
+			format:   "%s",
+			expected: "0",
+		},
+		{
+			time:     time.Date(2022, 2, 28, 12, 34, 56, 789000000, time.Local),
+			format:   "%c",
+			expected: "Mon Feb 28 12:34:56 2022",
+		},
+		{
+			time:     time.Date(2021, 10, 10, 10, 10, 10, 10000000, time.UTC),
+			format:   "%x %X %N %Q",
+			expected: "10/10/21 10:10:10 010000000 10",
+		},
+		{
+			time:     time.Date(1998, 5, 17, 0, 0, 0, 0, time.UTC),
+			format:   "%V",
+			expected: "20",
+		},
+		{
+			time:     time.Date(2000, 1, 2, 3, 4, 5, 678000000, time.UTC),
+			format:   "%Y-%m-%d %H:%M:%S%f",
+			expected: "2000-01-02 03:04:05.678000",
+		},
+		{
+			time:     time.Date(2000, 12, 25, 23, 59, 59, 999000000, time.UTC),
+			format:   "%c",
+			expected: "Mon Dec 25 23:59:59 2000",
+		},
+	}
+
+	for _, c := range cases {
+		formattedTime := formatTime(c.time, c.format)
+		if formattedTime != c.expected {
+			t.Errorf("Test failed for time: %v, format: %s, expected: %s, got: %s",
+				c.time, c.format, c.expected, formattedTime)
+		}
+	}
+}
+
+func TestParseTime(t *testing.T) {
+	tests := []struct {
+		dateStr  string
+		format   string
+		expected time.Time
+		err      bool
```

Comment:
nit: rename `err` to `shouldError` or similar so that from the name it's clear that it's a `bool` rather than an `error`